src/Orc.CsvTextEditor/Extensions/TextEditorExtensions.cs

```diff
@@ -38,5 +40,57 @@ public static void SetCaretToSpecificLineAndColumn(this TextEditor textEditor, i
             var newCaretOffset = offset + columnOffset;
             textEditor.CaretOffset = maxCaretOffset > newCaretOffset ? newCaretOffset : maxCaretOffset;
         }
+
+        public static List<ICompletionData> GetCompletionDataForText(this TextEditor textEditor, string autocompletionText, int columnIndex, int[][] scheme)
+        {
+            var textDocument = textEditor.Document;
+            var offset = textEditor.CaretOffset;
+
+            var affectedLocation = textDocument.GetLocation(offset);
+
+            var text = textEditor.Text;
+            var lines = textDocument.Lines;
+
+            var data = new List<ICompletionData>();
+            for (var i = 1; i < lines.Count; i++)
+            {
+                var line = lines[i];
+                if (line.Length == 0)
+                {
+                    continue;
+                }
+
+                var lineScheme = scheme[i];
+
+                var columnWidth = lineScheme[columnIndex] - 1;
+                var columnStart = lineScheme.Take(columnIndex).Sum();
+
+                var lineOffset = line.Offset;
+                var columnOffset = lineOffset + columnStart;
+
+                var columnChunk = text.Substring(columnOffset, columnWidth);
+                var words = columnChunk.Split(null);
+                var currentWord = text.GetWordFromOffset(offset - 1);
+                foreach (var word in words)
+                {
+                    if (string.Equals(currentWord, word))
+                    {
+                        continue;
+                    }
+
+                    if (data.Any(x => string.Equals(x.Text, word)))
+                    {
+                        continue;
+                    }
+
+                    if (word.ToLower().StartsWith(autocompletionText.ToLower()))
```

Comment:
you probably have to convert `autocompletionText` to lower only once, insted of doing it each time in loop